beacon-chain/sync/validate_beacon_attestation.go

```diff
@@ -94,76 +96,103 @@ func (s *Service) validateCommitteeIndexBeaconAttestation(ctx context.Context, p
 		return pubsub.ValidationIgnore
 	}
 
-	if err := s.chain.VerifyLmdFfgConsistency(ctx, att); err != nil {
```

Comment:
Shifting this check to later opens up a DOS vector as any attestation can be used to retrieve a far back pre-state. 

```
	preState, err := s.chain.AttestationPreState(ctx, att)
```
We retrieve this without checking the attestations  consistency with regards to the last finalized checkpoint. So an attestation referencing a target 1000 epochs back will be able to retrieve the prestate.

Reply:
Looking again, we need to shift this check before a prestate fetch.
https://github.com/prysmaticlabs/prysm/pull/7834/files/6b17fb83775ae0c1bcc593d06654df74596ef70b#diff-7bac03b93caca27e770ceadaedfc9084d7dcdb81c25d317fa823c86484274ffaR191